python/ray/_private/node.py

```diff
@@ -1001,12 +1005,7 @@ def _prepare_socket_file(self, socket_path: str, default_prefix: str):
             else:
                 try_to_create_directory(os.path.dirname(socket_path))
 
-            # Check socket path length to make sure it's short enough
-            maxlen = (104 if is_mac else 108) - 1  # sockaddr_un->sun_path
-            if len(result.split("://", 1)[-1].encode("utf-8")) > maxlen:
-                raise OSError(
-                    f"AF_UNIX path length cannot exceed {maxlen} bytes: {result!r}"
-                )
+            validate_socket_filepath(result.split("://", 1)[-1])
```

Comment:
Do we need to keep `.encode("utf-8")`?

Reply:
Do we want to use non-ASCII chracters for filename? If we only use ASCII characters for filename, then the length of string and the length of the encoded bytes will be the same.

Reply:
I don't know the history, I think it's safer to keep it.

Reply:
Added back `.encode` inside `validate_socket_filepath`.